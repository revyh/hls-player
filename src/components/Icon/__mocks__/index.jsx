import React from 'react';

export default function Icon(props) {
  return (
    <button
      className={props.className}
      onClick={props.onClick}
    >
      {props.symbol}
    </button>
  );
}
