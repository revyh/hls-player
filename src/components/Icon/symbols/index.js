const context = require.context('.', false, /\.svg$/);

export default context.keys().reduce(
  (result, symbol) => {
    const symbolName = (/\.\/(.*)\.svg$/).exec(symbol)[1];

    result[symbolName] = `#${symbolName}`;
    return result;
  },
  {},
);
