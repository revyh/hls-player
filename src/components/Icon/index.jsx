import React from 'react';
import symbols from './symbols';

export default function Icon(props) {
  const {className = '', symbol, onClick} = props;

  return (
    <svg className={className} onClick={onClick}>
      <use xlinkHref={symbols[symbol]} />
    </svg>
  );
}
