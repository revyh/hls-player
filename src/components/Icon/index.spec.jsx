import React from 'react';
import {shallow} from 'enzyme';
import Icon from '.';

jest.mock('./symbols', () => ({someSymbol: '/icons.svg#someSymbol'}));

it('should render', () => {
  expect(shallow(<Icon symbol="someSymbol" />)).toMatchSnapshot();
});

it('should trigger \'onClick\' event handler', () => {
  const mockOnClick = jest.fn();
  const icon = shallow(
    <Icon symbol="someSymbol" onClick={mockOnClick} />,
  );

  icon.simulate('click');

  expect(mockOnClick).toHaveBeenCalled();
});
