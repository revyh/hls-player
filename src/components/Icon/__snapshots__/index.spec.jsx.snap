// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<svg
  className=""
>
  <use
    xlinkHref="/icons.svg#someSymbol"
  />
</svg>
`;
